### ISSUE TITLE
How to build?

### ISSUE THREAD

**Issue by JamesDunne** [raysan5/raygui#19]
There appears to be no build system here unlike `raylib` which I've already compiled into a SHARED library my the Raspberry Pi 3B.

How does one build this library or even just one of the provided examples?

I've tried a very basic `gcc -I/home/pi/raylib/src controls_review.c` (so it can find `raylib.h`) and just get a ton of errors. What else do I need to `-D`efine?

**Comment by raysan5** [comment]
Actually, raygui is intended to be just manually added to your raylib project. Just copy the [raygui.h](https://github.com/raysan5/raygui/blob/master/src/raygui.h) file into your project folder and include it using:
```c
#define RAYGUI_IMPLEMENTATION
#include "raygui.h"
```
Those two lines, include the library header + implementation into your code. Super simple. 

To build the example, you can use the same method used to build any [raylib example](https://github.com/raysan5/raylib/wiki/Working-on-Raspberry-Pi#compiling-raylib-examples).

Lately we're working hard on raygui... but I agree there is a bit lack of documentation.

**Comment by JamesDunne** [comment]
It was failing to build on Raspberry Pi Linux because of the `__declspec(dllextern)` in `RAYGUIDEF`. After removing that, the examples compiled smoothly.

**Comment by raysan5** [comment]
oh, I see! Some user added it to build on MSVC I think... Thanks for reporting! 

Actually, I haven't tested raygui on RPI just yet...

**Comment by JamesDunne** [comment]
No worries. I got it figured out after an hour or so. The `controls_review.c` example worked well on the RPI 3B with the official 7" touchscreen. I just redefined it to 800x480 instead of 800x600, and one or two controls overlapped, but all in all it was functional. Great work!

**Comment by raysan5** [comment]
oh! that's great! Thanks for the feedback! :D

`controls_review.c` is the latest test bench, controls overlapping was for testing... the rest of the examples need to be reviewed... well, several things need review... keep working on it...

Please, keep me updated with your progress!

**Comment by JamesDunne** [comment]
Will do. I'm planning to build a shared library on the RPI to use with the C# bindings (raylib-cs).

![20181024_143111](https://user-images.githubusercontent.com/538152/47462071-faadbb00-d7a7-11e8-834c-d54915e53494.jpg)

Does that appear to be correct? I haven't even run the controls_review test on a desktop yet for a baseline lol.

**Comment by raysan5** [comment]
Yes, it is, same as desktop, well, a couple of controls are out of the window... Actually, it's a quite ugly testing sample... sorry...

Wow! `raylib-cs` running on the RPI? Is that possible? what about performance?

**Comment by JamesDunne** [comment]
Should be possible as long as I have a .so to PInvoke to. Not sure about performance yet which is why I wanted to try it out first :D
